opnode/rollup/driver/driver.go

```diff
@@ -21,23 +22,52 @@ type BatchSubmitter interface {
 	Submit(config *rollup.Config, batches []*derive.BatchData) (common.Hash, error)
 }
 
+type Downloader interface {
+	// FetchL1Info fetches the L1 header information corresponding to a L1 block ID
+	FetchL1Info(ctx context.Context, id eth.BlockID) (derive.L1Info, error)
+	// FetchReceipts of a L1 block
+	FetchReceipts(ctx context.Context, id eth.BlockID) ([]*types.Receipt, error)
+	// FetchTransactions from the given window of L1 blocks
+	FetchTransactions(ctx context.Context, window []eth.BlockID) ([]*types.Transaction, error)
+}
+
+type BlockPreparer interface {
```

Comment:
Can't we just call it the `ExecutionEngine`? Or maybe just `Engine`?

Reply:
Renamed to engine